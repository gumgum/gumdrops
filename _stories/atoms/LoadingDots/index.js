import React from 'react';
import { boolean, text, object } from '@storybook/addon-knobs';
import { optionalSelect } from '../../../components/utils/optionalSelect';

import readme from './README.md';
import LoadingDots from '../../../components/atoms/LoadingDots';

const sizeOptions = {
    sm: 'sm',
    lg: 'lg',
    '': 'No Value'
};

const component = () => (
    <LoadingDots
        size={optionalSelect('Size', sizeOptions, '')}
        whiteDots={boolean('White Dots', false)}
        className={text('ClassName', '-p-a-5')}
        style={object('Styles', {})}
    />
);

export default [readme, component];
