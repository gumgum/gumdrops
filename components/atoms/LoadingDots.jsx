import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const LoadingDots = ({ whiteDots, size, className, style }) => {
    const baseClass = 'gds-loading__dot';
    const dotClasses = cx(baseClass, className, {
        [`${baseClass}--${size}`]: size,
        [`${baseClass}--white`]: whiteDots
    });

    return (
        <div style={style} className="gds-loading">
            <div className={dotClasses} />
        </div>
    );
};

LoadingDots.displayName = 'LoadingDots';

LoadingDots.defaultProps = {
    whiteDots: false
};

LoadingDots.propTypes = {
    whiteDots: PropTypes.bool,
    /** sm, lg */
    style: PropTypes.object,
    size: PropTypes.oneOf(['sm', 'lg']),
    className: PropTypes.string
};

export default LoadingDots;
