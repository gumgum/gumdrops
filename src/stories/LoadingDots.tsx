import React from 'react';
import cx from 'classnames';
import { Sizes } from 'types';

export interface LoadingDotsProps extends React.HTMLAttributes<Element> {
    whiteDots?: boolean;
    size: Sizes;
}

const baseClass = 'gds-loading__dot';

export const LoadingDots: React.FC<LoadingDotsProps> = ({ whiteDots, size, ...otherProps }) => {
    const dotClasses = cx(baseClass, {
        [`${baseClass}--${size}`]: size,
        [`${baseClass}--white`]: whiteDots
    });

    return (
        <div {...otherProps}>
            <div className="gds-loading">
                <div className={dotClasses} />
            </div>
        </div>
    );
};
