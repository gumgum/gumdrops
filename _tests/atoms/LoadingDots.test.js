import renderer from 'react-test-renderer';
import React from 'react';
import LoadingDots from '../../components/atoms/LoadingDots';

const defaultProps = {
    className: 'cool-class',
    size: 'lg',
    style: { width: '100px' }
};

test('Expect <LoadingDots> to render properly', () => {
    const tree = renderer.create(<LoadingDots {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Expect <LoadingDots> to render properly with empty', () => {
    const props = {
        ...defaultProps,
        isWhite: true
    };

    const tree = renderer.create(<LoadingDots {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
});
