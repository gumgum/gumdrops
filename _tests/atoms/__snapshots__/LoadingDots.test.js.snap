// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <LoadingDots> to render 1`] = `
<LoadingDots
  className="cool-class"
  size="lg"
  style={
    Object {
      "width": "100px",
    }
  }
  whiteDots={false}
>
  <div
    className="gds-loading"
    style={
      Object {
        "width": "100px",
      }
    }
  >
    <div
      className="gds-loading__dot cool-class gds-loading__dot--lg"
    />
  </div>
</LoadingDots>
`;

exports[`Expect <LoadingDots> to render as white 1`] = `
<LoadingDots
  className="cool-class"
  isWhite={true}
  size="lg"
  style={
    Object {
      "width": "100px",
    }
  }
  whiteDots={false}
>
  <div
    className="gds-loading"
    style={
      Object {
        "width": "100px",
      }
    }
  >
    <div
      className="gds-loading__dot cool-class gds-loading__dot--lg"
    />
  </div>
</LoadingDots>
`;
